src/handler.ts

```diff
@@ -121,9 +129,9 @@ abstract class AbstractFileHandler implements IMessageFilter {
     if (this.widgets.indexOf(widget) === -1) {
       return;
     }
-    let path = AbstractFileHandler.pathProperty.get(widget);
-    return this.getState(widget).then(model => {
-      return this.manager.save(model.path, model);
+    let model = AbstractFileHandler.modelProperty.get(widget);
+    return this.getState(widget).then(contents => {
+      return this.manager.save(model.path, contents);
```

Comment:
Nice, I like how this doesn't let the widget change the filepath by returning a different state.  Good security catch.